semantics/executable-spec/src/Control/Iterate/SetAlgebra.hs

```diff
@@ -229,6 +230,12 @@ biMapFromList comb xs = foldr addEntry biMapEmpty xs
             Just oldv -> MkBiMap (addkv (k,v) forward comb) (insertBackwards oldv newv k backward)
                where newv = comb oldv v
 
+biMapFromAscDistinctList ::
+  (Ord k, Ord v) => [(k, v)] -> BiMap v k v
+biMapFromAscDistinctList xs = MkBiMap bmForward bmBackward
+  where
+    bmForward = Map.fromDistinctAscList xs
+    bmBackward = foldr (uncurry $ flip addBack) Map.empty xs
```

Comment:
I'm wondering how this compares to `foldl' (uncurry addBack) Map.empty xs`.

Reply:
I think it is worth adding to the benchmark a try of the foldl'  way. Being strict and a left fold should make things faster.